src/chainstate/stacks/address.rs

```diff
@@ -34,156 +34,62 @@ use chainstate::stacks::{
     }
 
-    pub fn is_mainnet(&self) -> bool {
-        match self.version {
-            C32_ADDRESS_VERSION_MAINNET_MULTISIG | C32_ADDRESS_VERSION_MAINNET_SINGLESIG => true,
-            C32_ADDRESS_VERSION_TESTNET_MULTISIG | C32_ADDRESS_VERSION_TESTNET_SINGLESIG => false,
-            _ => false,
-        }
-    }
-
-    pub fn burn_address(mainnet: bool) -> StacksAddress {
-        StacksAddress {
-            version: if mainnet {
-                C32_ADDRESS_VERSION_MAINNET_SINGLESIG
-            } else {
-                C32_ADDRESS_VERSION_TESTNET_SINGLESIG
-            },
-            bytes: Hash160([0u8; 20]),
-        }
-    }
-
-    pub fn from_burnchain_signer(o: &BurnchainSigner, mainnet: bool) -> Option<StacksAddress> {
-        let version = if mainnet {
-            match &o.hash_mode {
-                AddressHashMode::SerializeP2PKH => C32_ADDRESS_VERSION_MAINNET_SINGLESIG,
-                AddressHashMode::SerializeP2SH
-                | AddressHashMode::SerializeP2WPKH
-                | AddressHashMode::SerializeP2WSH => C32_ADDRESS_VERSION_MAINNET_MULTISIG,
-            }
-        } else {
-            match &o.hash_mode {
-                AddressHashMode::SerializeP2PKH => C32_ADDRESS_VERSION_TESTNET_SINGLESIG,
-                AddressHashMode::SerializeP2SH
-                | AddressHashMode::SerializeP2WPKH
-                | AddressHashMode::SerializeP2WSH => C32_ADDRESS_VERSION_TESTNET_MULTISIG,
-            }
-        };
-
-        StacksAddress::from_public_keys(version, &o.hash_mode, o.num_sigs, &o.public_keys)
-    }
-
-    /// Generate an address from a given address hash mode, signature threshold, and list of public
-    /// keys.  Only return an address if the combination given is supported.
-    /// The version is may be arbitrary.
-    pub fn from_public_keys(
-        version: u8,
-        hash_mode: &AddressHashMode,
-        num_sigs: usize,
-        pubkeys: &Vec<StacksPublicKey>,
-    ) -> Option<StacksAddress> {
-        // must be sufficient public keys
-        if pubkeys.len() < num_sigs {
-            return None;
-        }
-
-        // address hash mode must be consistent with the number of keys
-        match *hash_mode {
-            AddressHashMode::SerializeP2PKH | AddressHashMode::SerializeP2WPKH => {
-                // must be a single public key, and must require one signature
-                if num_sigs != 1 || pubkeys.len() != 1 {
-                    return None;
-                }
-            }
-            _ => {}
-        }
-
-        // if segwit, then keys must all be compressed
-        match *hash_mode {
-            AddressHashMode::SerializeP2WPKH | AddressHashMode::SerializeP2WSH => {
-                for pubkey in pubkeys {
-                    if !pubkey.compressed() {
-                        return None;
-                    }
-                }
+    fn to_bitcoin_tx_out(&self, value: u64) -> TxOut {
+        let btc_version = to_b52_version_byte(self.version)
```

Comment:
Do you mean `to_b58_version_byte`?

Reply:
I think this method is inappropriately named, repo-wide.

Reply:
Yes, this is just inappropriately named in the base branch -- I can rename this method in this refactor.